settings/rust-analyzer.vim

```diff
@@ -13,9 +13,19 @@ augroup vim_lsp_settings_rust_analyzer
       \ 'blocklist': lsp_settings#get('rust-analyzer', 'blocklist', []),
       \ 'config': lsp_settings#get('rust-analyzer', 'config', lsp_settings#server_config('rust-analyzer')),
       \ 'workspace_config': lsp_settings#get('rust-analyzer', 'workspace_config', {}),
+      \ 'capabilities': lsp_settings#get('rust-analyzer', 'capabilities', {'window': {'workDoneProgress': v:true }}),
       \ 'semantic_highlight': lsp_settings#get('rust-analyzer', 'semantic_highlight', {}),
       \ }
   autocmd User lsp_setup call s:register_command()
+
+  call lsp#callbag#pipe(
```

Comment:
This will register the same callbag twice. One for clangd and one for rust analyzer.

Two options.
1. Also check for `has_key(x, 'server') && x['server'] ==# 'rust-analyzer'`
2. Always listen to to `$/progress` no matter what the langserver is either during plugin load. You can also check for `g:lsp_work_done_progress_enabled`.
3. Always listen to to `$/progress` no matter what the langserver but explicitly subscribe once using callbag share. 

2. will most likely be the simpler version to implement if you are not familiar with 3.